java-checks/src/test/files/checks/DisallowedClassCheckRegex.java

```diff
@@ -0,0 +1,19 @@
+class A {
+  private String disallowed; // Noncompliant [[sc=11;ec=17]]
+  private Integer disallowed2; // Noncompliant [[sc=11;ec=18]]
+  private java.util.Date date;
+  String method1() { // Noncompliant [[sc=3;ec=9]]
+  }
+void method(String param) { // Noncompliant [[sc=13;ec=19]]
+  B var = new B(); // Compliant, no subtypes checked
+  String str = new String(); // Noncompliant [[sc=3;ec=9]]
+  System.out.println(var);
+  var.doSomething(new Integer(1)); // Noncompliant  [[sc=19;ec=33]]
+}
```

Comment:
You may want to add a case where we are not able to resolve the type (for example a method returning an Unknown type, not defined anywhere or not imported)